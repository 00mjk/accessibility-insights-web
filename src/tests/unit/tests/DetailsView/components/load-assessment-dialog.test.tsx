// // Copyright (c) Microsoft Corporation. All rights reserved.
// // Licensed under the MIT License.
// import { UrlParser } from 'common/url-parser';
// import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
// import {
//     LoadAssessmentDialog,
//     LoadAssessmentDialogDeps,
//     LoadAssessmentDialogProps,
// } from 'DetailsView/components/load-assessment-dialog';
// import { shallow } from 'enzyme';
// import * as React from 'react';
// import { It, Mock, MockBehavior, Times } from 'typemoq';

// describe('LoadAssessmentDialog', () => {
//     const detailsViewActionMessageCreatorMock = Mock.ofType(DetailsViewActionMessageCreator);
//     const urlParserMock = Mock.ofType(UrlParser, MockBehavior.Strict);

//     const deps = {
//         detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object,
//     } as LoadAssessmentDialogDeps;
//     const props = { deps } as LoadAssessmentDialogProps;

//     it('should render properly', () => {
//         const rendered = shallow(<LoadAssessmentDialog {...props} />);

//         expect(rendered.getElement()).toMatchSnapshot();
//     });

//     it('should render null when prevTab does not exists', () => {
//         prevTab => {
//             const detailsViewActionMessageCreatorMock = Mock.ofType(
//                 DetailsViewActionMessageCreator,
//             );

//             const newTab = {
//                 id: 111,
//                 url: 'https://www.def.com',
//                 title: 'test title',
//             };

//             urlParserMock
//                 .setup(urlParserObject => urlParserObject.areURLsEqual(It.isAny(), newTab.url))
//                 .returns(() => true)
//                 .verifiable(Times.never());

//             // const loadAssessmentDialogProps: LoadAssessmentDialogProps = {
//             //     deps: {
//             //         urlParser: urlParserMock.object,
//             //         detailsViewActionMessageCreator: detailsViewActionMessageCreatorMock.object,
//             //     },
//             //     prevTab: prevTab,
//             //     newTab: newTab,
//             // };

//             //const wrapper = shallow(<LoadAssessmentDialog {...loadAssessmentDialogProps} />);

//             // expect(wrapper.find(Dialog).exists()).toBe(false);
//             // urlParserMock.verifyAll();
//         }
// });


// });
