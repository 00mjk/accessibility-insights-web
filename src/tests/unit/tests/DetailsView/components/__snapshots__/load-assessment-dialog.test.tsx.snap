// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadAssessmentDialog should render properly 1`] = `
<ChangeAssessmentDialog
  deps={
    Object {
      "detailsViewActionMessageCreator": proxy {
        "___id": "BCDF5CE5-F0DF-40B7-8BA0-69DF395033C8",
        "addPathForValidation": [Function],
        "cancelStartOver": [Function],
        "cancelStartOverAllAssessments": [Function],
        "changeAssessmentVisualizationState": [Function],
        "changeManualRequirementStatus": [Function],
        "changeManualTestStatus": [Function],
        "clearPathSnippetData": [Function],
        "closePreviewFeaturesPanel": [Function],
        "closeScopingPanel": [Function],
        "closeSettingsPanel": [Function],
        "continuePreviousAssessment": [Function],
        "copyIssueDetailsClicked": [Function],
        "dispatcher": undefined,
        "editFailureInstance": [Function],
        "leftNavPanelExpanded": [Function],
        "loadAssessment": [Function],
        "removeFailureInstance": [Function],
        "rescanVisualization": [Function],
        "setAllUrlsPermissionState": [Function],
        "setFeatureFlag": [Function],
        "startOverAllAssessments": [Function],
        "switchToTargetTab": [Function],
        "telemetryFactory": undefined,
        "undoManualRequirementStatusChange": [Function],
        "undoManualTestStatusChange": [Function],
      },
    }
  }
  dialogContentTitle="Assessment in progress"
  dialogFirstText={
    <React.Fragment>
      Would you like to continue your current assessment or load the new Assessment?
    </React.Fragment>
  }
  dialogNoteText="If 'Continue previous' is selected, the assessment selected will not be loaded."
  dialogWarningText="If 'Load assessmentâ€™ is selected, all previous progress will be lost."
  divId="load-assessment-dialog-description"
  leftButtonOnClick={[Function]}
  leftButtonText="Continue previous"
  rightButtonOnClick={[Function]}
  rightButtonText="Load Assessment"
  subtitleAriaId="load-assessment-dialog-description"
/>
`;
